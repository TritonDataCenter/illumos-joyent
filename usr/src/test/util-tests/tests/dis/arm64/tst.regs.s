/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018 Joyent, Inc.
 */

/*
 * Test basic register naming
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	ADD W0, W1, W2
	ADD W3, W4, W5
	ADD W6, W7, W8
	ADD W9, W10, W11
	ADD W12, W13, W14
	ADD W15, W16, W17
	ADD W18, W19, W20
	ADD W21, W22, W23
	ADD W24, W25, W26
	ADD W27, W28, W29
	ADD W30, WSP, #1
	ADD X0, X1, X2
	ADD X3, X4, X5
	ADD X6, X7, X8
	ADD X9, X10, X11
	ADD X12, X13, X14
	ADD X15, X16, X17
	ADD X18, X19, X20
	ADD X21, X22, X23
	ADD X24, X25, X26
	ADD X27, X28, X29
	ADD X30, SP, #2

	ADD XZR, X0, X1
	ADD WZR, W1, W2
.size libdis_test, [.-libdis_test]
