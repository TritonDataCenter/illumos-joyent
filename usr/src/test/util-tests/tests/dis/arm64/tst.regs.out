    libdis_test:      20 00 02 0b  add w0, w1, w2
    libdis_test+0x4:  83 00 05 0b  add w3, w4, w5
    libdis_test+0x8:  e6 00 08 0b  add w6, w7, w8
    libdis_test+0xc:  49 01 0b 0b  add w9, w10, w11
    libdis_test+0x10: ac 01 0e 0b  add w12, w13, w14
    libdis_test+0x14: 0f 02 11 0b  add w15, w16, w17
    libdis_test+0x18: 72 02 14 0b  add w18, w19, w20
    libdis_test+0x1c: d5 02 17 0b  add w21, w22, w23
    libdis_test+0x20: 38 03 1a 0b  add w24, w25, w26
    libdis_test+0x24: 9b 03 1d 0b  add w27, w28, w29
    libdis_test+0x28: fe 07 00 11  add w30, wsp, #0x1
    libdis_test+0x2c: 20 00 02 8b  add x0, x1, x2
    libdis_test+0x30: 83 00 05 8b  add x3, x4, x5
    libdis_test+0x34: e6 00 08 8b  add x6, x7, x8
    libdis_test+0x38: 49 01 0b 8b  add x9, x10, x11
    libdis_test+0x3c: ac 01 0e 8b  add x12, x13, x14
    libdis_test+0x40: 0f 02 11 8b  add x15, x16, x17
    libdis_test+0x44: 72 02 14 8b  add x18, x19, x20
    libdis_test+0x48: d5 02 17 8b  add x21, x22, x23
    libdis_test+0x4c: 38 03 1a 8b  add x24, x25, x26
    libdis_test+0x50: 9b 03 1d 8b  add x27, x28, x29
    libdis_test+0x54: fe 0b 00 91  add x30, sp, #0x2
    libdis_test+0x58: 1f 00 01 8b  add xzr, x0, x1
    libdis_test+0x5c: 3f 00 02 0b  add wzr, w1, w2
